lua/orgmode/capture/init.lua

```diff
@@ -240,49 +223,27 @@ function Capture:_refile_content_with_fallback(opts)
       return false
     end
     opts.file = default_file
-    return self:_refile_to_end(opts)
+    return self:_refile_to(opts)
   end
 
   opts.file = valid_destinations[destination[1]]
   opts.headline = table.concat({ unpack(destination, 2) }, '/')
-  if not opts.headline or opts.headline == '' then
-    return self:_refile_to_end(opts)
-  end
-  return self:refile_to_headline(opts)
+  return self:_refile_to(opts)
 end
 
----@param opts CaptureOpts
-function Capture:refile_to_headline(opts)
-  local destination_file = Files.get(opts.file)
-  local headline
-  if opts.headline then
-    headline = destination_file:find_headline_by_title(opts.headline, true)
-
-    if not headline then
-      utils.echo_error("headline '" .. opts.headline .. "' does not exist in '" .. opts.file .. "'. Aborted refiling.")
-      return false
-    end
+---@param item Section
+---@param target_level integer
+---@param is_same_file boolean
+function Capture:_adapt_headline_level(item, target_level, is_same_file)
+  -- Refiling in same file just moves the lines from one position
+  -- to another,so we need to apply demote instantly
+  if target_level == 0 then
+    return item:promote(item.level - 1, true, not is_same_file)
+  elseif item.level <= target_level then
+    return item:demote(target_level - item.level + 1, true, not is_same_file)
+  else
+    return item:promote(item.level - target_level - 1, true, not is_same_file)
```

Comment:
Since we do early returns, I'd prefer having each if statement separately instead of elseif.

```suggestion
  if target_level == 0 then
    return item:promote(item.level - 1, true, not is_same_file)
  end
  if item.level <= target_level then
    return item:demote(target_level - item.level + 1, true, not is_same_file)
  end
  return item:promote(item.level - target_level - 1, true, not is_same_file)
```

Reply:
Done.